state/protocol/badger/mutator_test.go

```diff
@@ -544,45 +556,84 @@ func TestExtendBlockNotConnected(t *testing.T) {
 	})
 }
 
+// Test that seals are rejected if they correspond to ExecutionResults that are
+// not incorporated in blocks on this fork
+func TestExtendSealNoIncorporatedResult(t *testing.T) {
```

Comment:
This is a great test. Thank you. 

* consider Block `B0`
  - there are two Execution results for `B0`: `R_a` and `R_b`
  - block `B1` is a  descendant of `B0` and contains a `S_a` for an execution result `R_a`

In my mind, there are 3 distinct failure scenarios here: 
1.  _no_ execution result for block `B0` was included in the fork with head `B1`
2.  The execution result `R_b` for block `B0` was included in the fork with head `B1`. However, as `S_a` seals result `R_a` and not `R_b`, this is still an invalid extension.
3.  The execution result `R_a` was included in a _different_ fork than the one with head `B1` 

It would be great to have tests for all 3 cases.